inc/healthcheck/namespace.php

```diff
@@ -26,35 +26,75 @@ function bootstrap() {
 		return;
 	}
 
+	// Defer app level healthchecks until after plugins have laoded.
+	add_action( 'plugins_loaded', __NAMESPACE__ . '\\load_healthcheck', 100 );
+
+	// Run instance healthcheck immediately.
+	if ( strpos( $_SERVER['REQUEST_URI'], '/__instance_healthcheck' ) !== false ) {
+		output_page( run_instance_checks() );
+	}
+}
+
+/**
+ * Loads the application healthcheck.
+ *
+ * @return void
+ */
+function load_healthcheck() {
+	if ( defined( 'WP_INSTALLING' ) && WP_INSTALLING ) {
+		return;
+	}
+
+	Cavalcade\bootstrap();
+
 	if ( strpos( $_SERVER['REQUEST_URI'], '/__healthcheck' ) !== false ) {
 		output_page( run_checks() );
 		return;
 	}
+}
 
-	if ( strpos( $_SERVER['REQUEST_URI'], '/__instance_healthcheck' ) !== false ) {
-		output_page( run_instance_checks() );
+/**
+ * Check if a value is an error.
+ *
+ * @param mixed $value The variable to check.
+ * @return boolean
+ */
+function is_error( $value ) : bool {
+	if ( function_exists( 'is_wp_error' ) ) {
+		return is_wp_error( $value );
 	}
+	return $value !== true;
 }
```

Comment:
`is_wp_error` is simple enough to replicate, maybe we should go for the instanceof check here ?

```suggestion
function is_error( $value ) : bool {
	return $value instanceof WP_Error;
}
```

Reply:
I was thinking the same, but presumably the class isn't loaded?

Reply:
I think bunch of the checks are already assuming it exists, eg: `run_mysql_healthcheck`. @roborourke can you clarify the use case for this change ?

Reply:
I went for this approach in case we needed to add additional checks to the instance healthcheck in future. If we do those can't return a `WP_Error` but might need some sort of failed status still. Given our other healthchecks return `true` or a `WP_Error` I felt this was the most straight forward to account for alternative healthcheck return values before WP has loaded.

Reply:
Also not sure if the `WP_Error` class would have to exist to be able to do an instanceof check against it 🤔

Reply:
Fair! I take back my suggestion then.

Reply:
TIL that can be done, the class doesn't need to exist. Still though, needs to fall back to checking if the value is `true` or not otherwise an early health check could return false and still count as a pass.